components/net/image_cache_task.rs

```diff
@@ -462,6 +485,15 @@ fn load_image_data(url: Url, resource_task: ResourceTask) -> Result<Vec<u8>, ()>
                 return Ok(image_data);
             }
             Done(Err(..)) => {
+                // Failure to load the requested image will return the
+                // placeholder instead. In case it failed to load at init(),
+                // we still recover and return Err() but nothing will be drawn.
+                if placeholder.len() != 0 {
+                    debug!("image_cache_task: failed to load, use placeholder instead.");
+                    return Ok(placeholder.clone());
+                } else {
+                    debug!("image_cache_task: invalid placeholder.");
+                }
```

Comment:
This whole thing could alternatively be written like:

```
if ... {
    ...
    Ok()
} else {
   ...
   Err()
}
```

Instead of using any explicit `return`s